Neos.ContentRepository.Core/Classes/Feature/Common/TetheredNodeInternals.php

```diff
@@ -136,12 +136,10 @@ protected function createEventsForMissingTetheredNode(
                 );
             }
 
-            $childNodeSource = null;
-            foreach ($childNodeAggregate->getNodes() as $node) {
-                $childNodeSource = $node;
-                break;
-            }
-            /** @var Node $childNodeSource Node aggregates are never empty */
+            $rootDimensionSpacePoints = $this->interDimensionalVariationGraph->getRootGeneralizations();
+            $arbitraryRootDimensionSpacePoint = array_shift($rootDimensionSpacePoints);
+            $childNodeSource = $childNodeAggregate->getNodeByCoveredDimensionSpacePoint($arbitraryRootDimensionSpacePoint);
+
             return $this->createEventsForVariations(
                 $parentNodeAggregate->contentStreamId,
                 $childNodeSource->originDimensionSpacePoint,
```

Comment:
im actually not really sure what we do here, and interestingly the whole codepath is not covered by any of the `ContentRepository.BehavioralTests` tests. Throwing an exception for example here will not cause a testfailure.

So its really a good question what we are doing here and how this codepath might be tested 😅 ... we only seem to need a node to get its `originDimensionSpacePoint` ... maybe its equivalent to


```
$occupiedDimensionSpacePoints = $childNodeAggregate->occupiedDimensionSpacePoints->getPoints();
$arbitraryOccupiedDimensionSpacePoint = array_shift($occupiedDimensionSpacePoints);
```

but to be honest idk, and its kinda scary to not have a test for it :D

Reply:
With a little debugging i could came up with a test, but it seems the current state is not only hard to read but also broken, as the test fails :D

Reply:
Moved to https://github.com/neos/neos-development-collection/pull/4969